NickelHook.h

```diff
@@ -26,6 +26,7 @@ struct nh {
     struct nh_info  *info;
     struct nh_hook  *hook;  // pointer to the first element, NULL terminated
     struct nh_dlsym *dlsym; // pointer to the first element, NULL terminated
+    void (*uninstall)(); // optional, allow extra cleanup on uninstall
```

Comment:
Is there any point in this returning a non-void result? I'm not sure as there's anything else that can be done if an error occurs in the function.

Reply:
I think this should return a bool for success so NH can dump the logs if it fails. But, even if it returns false, NH will still continue with its own uninstall.

Reply:
Also, can you add a new line before it? I'm keeping the required options and each set of added options separated.